tests/NexusMods.DataModel.Tests/DataStoreTests.cs

```diff
@@ -33,13 +33,50 @@ public void CanGetAndSetHashedValues()
             Id = ModFileId.New(),
             Hash = Hash.Zero,
             From = new HashRelativePath((Hash)42L, default),
-            Size = (Size)42L,
+            Size = Size.From(42L),
             To = new GamePath(GameFolderType.Game, "test.foo")
         }.WithPersist(DataStore);
         foo.DataStoreId.ToString().Should().NotBeEmpty();
         DataStore.Get<FromArchive>(foo.DataStoreId).Should().NotBeNull();
     }
 
+    [Fact]
+    public void CompareAndSwapIsAtomic()
+    {
+        const int numThreads = 10;
```

Comment:
We should limit the number of threads using `Environment.ProcessorCount`. Hardcoding the number of threads might produce weird results in CI, which is often very limited.

Reply:
Something small to add here.
`Environment.ProcessorCount` is ideal, but some lower bound should probably also be established. The test would be moot if by some chance it ran in a single threaded environment.

IIRC GitHub Actions has 2 threads on Windows/Linux runners and 3 on OSX.

Reply:
The point here is to slam the method so we get wildly inconsistent values if the function isn't atomic, but we can probably get away with `Math.Min(4, Environment.ProcessorCount * 2)`